core/trie/trie.go

```diff
@@ -157,8 +190,13 @@ func (t *Trie) Get(key *felt.Felt) (*felt.Felt, error) {
 }
 
 // Put updates the corresponding `value` for a `key`
+//
+//nolint:gocyclo
 func (t *Trie) Put(key, value *felt.Felt) (*felt.Felt, error) {
-	// Todo: check key is not bigger than max key value for a trie height.
+	if key.Cmp(maxKey) > 0 {
+		return nil, &errMaxKey{key: key}
+	}
```

Comment:
you can extract this into a method and it should solve the gocylo issue

Reply:
I think we'll still need to check the output of the method, so the number of branches will remain the same. I'll think about how to cleanly refactor `Put` to avoid the gocyclo issue.